etl-pipeline/mappings/oclc_bib.py

```diff
@@ -2,130 +2,189 @@
-                return True
-        
-        return False
+    publication_date = dates.get('publicationDate')
+
+    if not publication_date:
+        return None
     
-    def _map_subjects(self, oclc_bib) -> list[str]:
-        return [
-            f"{subject_name}||{subject.get('vocabulary', '')}" 
-            for subject in oclc_bib.get('subjects', [])
-            if (subject_name := subject.get('subjectName', {}).get('text'))
-        ]
+    return [f'{publication_date}|publication_date']
+
+
+def _get_languages(oclc_bib):
+    languages = oclc_bib.get('language')
+
+    if not languages:
+        return None
     
-    def _map_authors(self, authors) -> Optional[list[str]]:
-        if not authors:
-            return None
-        
-        return [
-            f'{author_name}|||true' 
-            for author in authors 
-            if (author_name := self._get_contributor_name(author))
-        ]
+    item_language = languages.get('itemLanguage')
+
+    if not item_language:
+        return None
     
-    def _map_contributors(self, contributors) -> Optional[list[str]]:
-        if not contributors:
-            return None
-        
-        return [
-            f"{contributor_name}|||{', '.join(list(map(lambda relator: relator.get('term', ''), contributor.get('relators', []))))}"
-            for contributor in contributors 
-            if (contributor_name := self._get_contributor_name(contributor))
-        ]
+    return [f'||{item_language}']
+
+
+def _get_parts(oclc_bib):
+    digital_access_locations = oclc_bib.get('digitalAccessAndLocations')
+
+    if not digital_access_locations:
+        return None
+    
+    return [
+        str(Part(index=1, url=uri, source='oclc', file_type='text/html', flags=str(FileFlags(embed=True))))
+        for digital_access_location in digital_access_locations
+        if (uri := digital_access_location.get('uri'))
+    ]
+
+
+def _get_creators(oclc_bib):
+    if not oclc_bib.get('contributor'): 
+        return None
+
+    return list(
+        filter(
+            lambda creator: creator.get('secondName') or creator.get('firstName'), 
+            oclc_bib.get('contributor', {}).get('creators', [])
+        )
+    )
+
+def _get_authors(creators):
+    if not creators:
+        return None
+
+    return list(
+        filter(
+            lambda creator: creator.get('isPrimary', False) or _is_author(creator), 
+            creators
+        )
+    )
+
+def _get_contributors(creators):
+    if not creators:
+        return None
+
+    return list(
+        filter(
+            lambda creator: not creator.get('isPrimary', False) and not _is_author(creator), 
+            creators
+        )
+    )
+
+def _is_author(creator):
+    for role in set(map(lambda relator: relator.get('term', '').lower(), creator.get('relators', []))):
+        if 'author' in role.lower() or 'writer' in role.lower():
+            return True
+    
+    return False
```

Comment:
Might consider using the `any` function here (personal pref!).  And wrapping in a `set` is probably unnecessary since you're just short circuiting once you find a match?

Reply:
bloop - ya this was when my python skills were not up to par haaaa - i will circle back to this :D